src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp

```diff
@@ -0,0 +1,356 @@
+
+size_t jit_power_static_emitter::get_aux_gprs_count() const { return 2; }
+
+void jit_power_static_emitter::register_table_entries() {
+    push_arg_entry_of("power", dnnl::impl::float2int(power), true);
+    push_arg_entry_of("scale", dnnl::impl::float2int(scale), true);
+    push_arg_entry_of("shift", dnnl::impl::float2int(shift), true);
+}
+
+std::set<std::vector<element::Type>> jit_power_static_emitter::get_supported_precisions(const std::shared_ptr<ngraph::Node>& node) {
+    return {{element::f32, element::f32}};
+}
+
+void jit_power_static_emitter::emit_impl(const std::vector<size_t>& in_vec_idxs, const std::vector<size_t>& out_vec_idxs) const {
+    if (host_isa_ == dnnl::impl::cpu::aarch64::asimd) {
+        emit_isa<dnnl::impl::cpu::aarch64::asimd>(in_vec_idxs, out_vec_idxs);
+    } else {
+        IE_THROW() << "Can't create jit eltwise kernel";
+    }
+}
+
+template <dnnl::impl::cpu::aarch64::cpu_isa_t isa>
+void jit_power_static_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
+    if (exec_prc_ != ov::element::f32) {
+        IE_THROW() << "unsupported precision: " << exec_prc_;
+    }
+
+    using TReg = typename dnnl::impl::cpu::aarch64::cpu_isa_traits<isa>::TReg;
+    TReg aux = TReg(aux_vec_idxs[0]);
+    TReg dst = TReg(out_vec_idxs[0]);
+
+    bool get_from_dst = false;
+    const auto src = [&in_vec_idxs, &out_vec_idxs, &get_from_dst]() -> TReg {
+        return get_from_dst ? TReg(out_vec_idxs[0]) : TReg(in_vec_idxs[0]);
+    };
+
+    if (scale != 1.f) {
+        auto adr = table_val2("scale");
+        h->ld1r(aux.s, adr);
+        h->fmul(dst.s, src().s, aux.s);
+        get_from_dst = true;
+    }
+
+    if (shift != 0.f) {
+        auto adr = table_val2("shift");
+        h->ld1r(aux.s, adr);
+        h->fadd(dst.s, src().s, aux.s);
+        get_from_dst = true;
+    }
+
+    if (power == 0.f) {
+        h->fmov(dst.s, 1.);
+        return;
+    }
+
+    if (power == 1.f) {
+        if (!get_from_dst && (in_vec_idxs[0] != dst.getIdx())) {
+            h->uni_orr(dst, src(), src());
+        }
+        return;
+    }
+
+    if (std::floor(power) == power && power > 0) {
+        h->uni_orr(aux, src(), src());
+        h->fmov(dst.s, 1.);
+
+        auto current_power = static_cast<size_t>(power);
+        while (current_power > 0) {
+            if (current_power & 1) {
+                h->fmul(dst.s, dst.s, aux.s);
+            }
+            if (current_power > 1) {
+                h->fmul(aux.s, aux.s, aux.s);
+            }
+            current_power = current_power >> 1;
+        }
+    } else {
+        auto pow_f32_addr = reinterpret_cast<uintptr_t>(::powf);
+
+        Xbyak_aarch64::XReg func_reg(aux_gpr_idxs[0]);
+        h->mov(func_reg, pow_f32_addr);
+
+        Xbyak_aarch64::SReg s0(0);
+        Xbyak_aarch64::SReg s1(1);
+
+        for (auto i = 0; i < 4; i++) {
+            store_context();
```

Comment:
Do we really need to store and restore context on each iteration?

Reply:
done, thanks

Reply:
discussed online: fixed, thanks